tests/test_databases.py

```diff
@@ -301,10 +306,17 @@ def run_sync(*args, **kwargs):
             pk = await database.execute(query, values)
             assert isinstance(pk, int)
 
-            query = notes.select().where(notes.c.id == pk)
-            result = await database.fetch_one(query)
-            assert result["text"] == "example1"
-            assert result["completed"] == True
+            # Apparently for `aiopg` it's OID that will always 0 in this case
+            # As it's only one action within this cursor life cycle
+            # It's recommended to use the `RETURNING` clause
+            # For obtaining the record id
```

Comment:
This is a good example of where we ought to be returning a more structured type from `.execute` - that way we could have different drivers either support or not support `result.row_id`.

Reply:
Can we change our query here to make it consistently return the pk across all our implementations maybe?

Reply:
Yes, sounds reasonable indeed.
There're two ways though as I can see:
- provide pure driver experience with minimal overhead, hence it'll vary from driver to driver, but will keep it as simple as possible if it can matter, or
- provide unified API and implement the missing bits providing the missing functionality, but allocating the additional overhead

Reply:
Oh, just 25 secs later, ok, let's follow the second path, all in all, there's always a way to jump to the raw driver and get the full manual control over it.

Reply:
@tomchristie I'm thinking quite hard about it. There's a function in postgres - `lastval()` that can be executed to retrieve the value, but it's not reliable as there're circumstances where it can return the wrong value. The `returning` clause is the best way to do it, but we can't assume how the PK column is called and `returning *` won't be consistent with other back-ends.
The